File: build-a-gradebook-app.py
def get_average(scores):
    return sum(scores) / len(scores)

def get_grade(score):
    if score == 100:
        return "A+"
    elif score >= 90:
        return "A"
    elif score >= 80:
        return "B"
    elif score >= 70:
        return "C"
    elif score >= 60:
        return "D"
    else:
        return "F"

def has_passing_grade(score):
    return get_grade(score)!= "F"

def student_msg(total_scores, student_score):
    average = get_average(total_scores)
    grade = get_grade(student_score)
    passed = has_passing_grade(student_score)
    status = "passed" if passed else "failed"
    return f"Class average: {average}. Your grade: {grade}. You {status} the course."

File: build-a-gradebook-app.js
function getAverage(scores) {
    let sum =0;
    for (let i=0; i<scores.length; i++) {
        sum += scores[i];
    }
    return sum / scores.length;
}

function getGrade(score) {
  if (score === 100) return "A+";
  if (score >= 90) return "A";
  if (score >= 80) return "B";
  if (score >= 70) return "C";
  if (score >= 60) return "D";
  return "F";
}

function hasPassingGrade(score) {
    return getGrade(score) !== "F";
}

function studentMsg(totalScores, studentScore) {
    const average = getAverage(totalScores);
    const grade = getGrade(studentScore);
    const passed = hasPassingGrade(studentScore);

    if (passed) {
    return "Class average: " + average + ". Your grade: " + grade + ". You passed the course.";
  } else {
    return "Class average: " + average + ". Your grade: " + grade + ". You failed the course.";
  }
}

//variant

function studentMsg1(totalScores, studentScore) {
    const average = getAverage = getAverage(totalScores);
    const grade = getGrade(studentScore);
    const passed = hasPassingGrade(studentScore);
    
    const status = passed? "passed" : "failed";
    return `Class average: ${average}. Your grade: ${grade}. You ${status} the course.`;
}